### PR TITLE
Fix subscription hookup in LanguageServiceHost.OnProjectFactoryComple…

### LINKED ISSUE #1666: Language Service is busted with 1.1 and restore
Repro Steps:
1. Create a new console app
2. Add NuGet package
3. Add code using the new types
4. Remove the NuGet package

Expected:
Errors in the error list.

Actual:
No errors, types are still highlighted, intellisense still provides removed package.

Notes:
I can't repro with with tfm ~~`netstandard1.0`~~ `netcoreapp1.0`, only with ~~`netstandard1.1`~~ `netcoreapp1.1`.

### PR DESCRIPTION
**Pull request by mavasani**
…tedAsync

Do not hookup subscription to design time builds in OnProjectFactoryCompletedAsync, and only invoke InitializeAsync for evaluation subscription hooked up in this method. This way, we will only process project change subscription updates  after LanguageServiceHost has been initialized, and we have a valid active project configuration, not just the NoVariablesDefault fallback configuration.

Fixes #1666

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs [hunk 1]
```diff
@@ -115,14 +108,12 @@ Task ILanguageServiceHost.InitializeAsync(CancellationToken cancellationToken)
             return InitializeAsync(cancellationToken);
         }
 
-        private async Task OnProjectChangedAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> e, RuleHandlerType handlerType)
+        private async Task InitializeAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> e, RuleHandlerType handlerType)
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: Why are these parameters needed?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs [hunk 2]
```diff
@@ -85,17 +85,10 @@ private Task OnProjectFactoryCompletedAsync()
             using (_tasksService.LoadedProject())
             {
                 var watchedEvaluationRules = GetWatchedRules(RuleHandlerType.Evaluation);
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: Why do we care about the handlers in this case? We only care about TFM, right?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs (hunk 1 above)
Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: I'll remove them

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs (hunk 2 above)
Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: That's correct, I can change this to just ConfigurationGeneral.SchemaName rule.

**Comment by davkean** [comment]
Thinking about this - what exactly is the point of hooking the first evaluation anyway? Why don't you just Initialize from the FactoryCompleted callback? Every following evaluation after the first one is ignored and you aren't even using the results from the evaluation.

**Comment by davkean** [comment]
Actually thinking even more about this - does this work as expected? Weren't we relying that the time that the design-time build occured that the config had changed? Wouldn't this suffer from the same problem?

**Comment by mavasani** [comment]
@davkean I verified under the debugger that by the time we start to execute the core LS host initialization, the configuration is set correctly. Basically, I altered my changes locally to just return InitializeAsync() in OnProjectFactoryCompletedAsync and I noticed that the active project config is NoVariablesDefault at the time we invoke InitializeAsync, but is correctly set to Debug | AnyCPU when we create the workspace project context at this line: https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs#L241. Note that we force CPS to refresh the active project configuration at the line above.

**Comment by davkean** [comment]
But is that a timing thing? What is _actually_ causing us to switch configs by that time?

**Comment by mavasani** [comment]
Implementation of [RefreshActiveProjectConfigurationAsync](http://index/?leftProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&leftSymbol=m9lhzdwg23jn&file=VsUnconfiguredProjectIntegrationService.cs&line=270), which invokes [RefreshActiveProjectConfigurationAsync](http://index/?leftProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&leftSymbol=m9lhzdwg23jn&file=VsUnconfiguredProjectIntegrationService.cs&line=393) suggests that it will guarantee correct VS configuration will be loaded by CPS, but @lifengl to confirm.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs [hunk 3]
```diff
@@ -115,16 +102,6 @@ Task ILanguageServiceHost.InitializeAsync(CancellationToken cancellationToken)
...
-                return;
-
-            await InitializeAsync().ConfigureAwait(false);
-
-            await OnProjectChangedCoreAsync(e, handlerType).ConfigureAwait(false);
-        }
-
         private async Task OnProjectChangedCoreAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> e, RuleHandlerType handlerType)
         {
             await _tasksService.LoadedProjectAsync(async () =>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: We no longer bail on being disposed here anymore.

**Comment by davkean** [comment]
Other than my last comment, everything else looks good.

**Comment by mavasani** [comment]
@davkean I have addressed your feedback.

**Pull request merged**